azurelinuxagent/common/persist_firewall_rules.py

```diff
@@ -112,6 +117,13 @@ def setup(self):
             # In case of a failure, this would throw. In such a case, we don't need to try to setup our custom service
             # because on system reboot, all iptable rules are reset by firewalld.service so it would be a no-op.
             self._setup_permanent_firewalld_rules()
+
+            # Remove custom service if exists to avoid problems with firewalld
+            try:
+                fileutil.rm_files(*[self.get_service_file_path(), os.path.join(conf.get_lib_dir(), self.BINARY_FILE_NAME)])
```

Comment:
would this need a daemon-reload?

Reply:
No, I verified manually, it didnt complain for it (I just verified it in the one distro where we know firewalld is there - RHEL 8+)